### PR TITLE
Add Wrappers

### PR DESCRIPTION
**Pull request by boutinb**
Clear up also the QML Import statements

### CODE REVIEW

File: inst/qml/AnovaBayesian.qml [hunk 1]
```diff
@@ -16,10 +16,9 @@
 // <http://www.gnu.org/licenses/>.
 //
 
-import QtQuick			2.12
-import JASP.Controls	1.0
-import JASP.Widgets		1.0
-import JASP				1.0
```
Comment (vandenman) [inline inst/qml/AnovaBayesian.qml:22]: removed `JASP.Widgets`

File: inst/qml/AnovaRepeatedMeasures.qml [hunk 2]
```diff
@@ -16,10 +16,9 @@
 // <http://www.gnu.org/licenses/>.
 //
 
-import QtQuick			2.12
-import JASP.Controls	1.0
-import JASP.Widgets		1.0
-import JASP				1.0
```
Comment (vandenman) [inline inst/qml/AnovaRepeatedMeasures.qml:22]: removed `JASP.Widgets`

**Comment by vandenman** [comment]
Also, you need to change all unit tests:
https://github.com/jasp-stats/jaspAnova/blob/8c86260cc1936ba91f87efabdffbbce2366a51a2/tests/testthat/test-anova.R#L59

should become
```r
 results <- jaspTools::runAnalysis("AnovaInternal", "test.csv", options)
```

furthermore, I'm not sure if this will run inside JASP. Did you test it?

File: inst/qml/AnovaBayesian.qml (hunk 1 above)
Comment (boutinb) [reply inst/qml/AnovaBayesian.qml:22]: Yes JASP.Widgets should not be used by the Analysis Forms anymore. JASP.Controls should contain all controls needed for the forms. JASP.Widgets is for extra widgets used by JASP Desktop.

**Comment by boutinb** [comment]
> Also, you need to change all unit tests:

I had a look, it is possible to change all runAnalysis with the internal function, but I think it becomes unclear for the developer when he has to call the function or the internal function. Later when the QMLComponents will be included in jaspBase, the unit test will be able to ask directly which function it has to call. So for the time being, we could just modify the findCorrectFunction in utils.R, like this:
```
findCorrectFunction <- function(funName) {
  modulePath <- getModulePathFromRFunction(funName)
  if (is.null(modulePath))
    stop("Could not locate the module location for `", funName, "`")
  
  if (!endsWith(funName, "Internal")) {
    funNameInternal <- paste0(funcName, "Internal")
    if (rFunctionExistsInModule(funNameInternal, modulePath))
      funName <- funNameInternal
  }

  return(paste(getModuleName(modulePath), funName, sep = "::"))
}
```

What do you think?

**Comment by vandenman** [comment]
But then developers should know that they have to create/ adjust `AnovaInternal` but call `Anova` without the `Internal`? That's also a bit unintuitive. In particular because most of the time, the wrapper for the internal function won't exist yet. I'd prefer just renaming the calls in the unit tests to the internal functions. It would be relatively easy to do that using find all and replace?

Oh and btw, the changes functions names also need to be in Upgrades.qml :upside_down_face:

**Comment by boutinb** [comment]
> But then developers should know that they have to create/ adjust `AnovaInternal` but call `Anova` without the `Internal`? That's also a bit unintuitive. In particular because most of the time, the wrapper for the internal function won't exist yet. I'd prefer just renaming the calls in the unit tests to the internal functions. It would be relatively easy to do that using find all and replace?
> 
> Oh and btw, the changes functions names also need to be in Upgrades.qml 🙃

The change I propose in utils.R make it possible to call either Anova or AnovaInternal. I was also thinking of the Developer: if I see the unit tests, then some functions have to be called with the Anova (like jaspTools::analysisOptions), other with AnovaInternal (like jaspTools:: runAnalysis), and it is not really clear why. That's why I thought to make it possible to call everything with just Anova (but he can still call the jaspTools:: runAnalysis with AnovaInternal).
This PR does not need an upgrade: the analysis name is still Anova: just JASP decides to call AnovaInternal or Anova.

**Comment by boutinb** [comment]
I have made the changes in jaspTools. I could test with the unit test on jaspDescriptives and this works. So I think we can merge this PR (the unit tests of ANOVA still fails but because of other problems).

**Pull request merged**